packages/by-name/kata/kata-runtime/0007-genpolicy-enable-sysctl-checks.patch

```diff
@@ -0,0 +1,222 @@
++        print("allow_linux_sysctl: i_name =", i_name, "i_val =", i_val)
++        p_sysctl[i_name] == i_val
++    }
++    print("allow_linux_sysctl: true")
++}
++
+ # Check the consistency of the input "io.katacontainers.pkg.oci.bundle_path"
+ # and io.kubernetes.cri.sandbox-id" values with other fields.
+ allow_by_bundle_or_sandbox_id(p_oci, i_oci, p_storages, i_storages) {
+diff --git a/src/tools/genpolicy/src/containerd.rs b/src/tools/genpolicy/src/containerd.rs
+index 075fced5b..2922ea0ab 100644
+--- a/src/tools/genpolicy/src/containerd.rs
++++ b/src/tools/genpolicy/src/containerd.rs
+@@ -3,6 +3,8 @@
+ // SPDX-License-Identifier: Apache-2.0
+ //
+ 
++use std::collections::HashMap;
++
+ use crate::policy;
+ 
+ // Default process field from containerd.
+@@ -153,6 +155,7 @@ pub fn get_linux(privileged_container: bool) -> policy::KataLinux {
+                 "/proc/sysrq-trigger".to_string(),
+             ],
+             Devices: vec![],
++            Sysctl: HashMap::new(),
+         }
+     } else {
+         policy::KataLinux {
+@@ -160,6 +163,7 @@ pub fn get_linux(privileged_container: bool) -> policy::KataLinux {
+             MaskedPaths: vec![],
+             ReadonlyPaths: vec![],
+             Devices: vec![],
++            Sysctl: HashMap::new(),
+         }
+     }
+ }
+diff --git a/src/tools/genpolicy/src/pod.rs b/src/tools/genpolicy/src/pod.rs
+index 4fda02916..ef5dece58 100644
+--- a/src/tools/genpolicy/src/pod.rs
++++ b/src/tools/genpolicy/src/pod.rs
+@@ -21,6 +21,7 @@ use log::{debug, warn};
+ use protocols::agent;
+ use serde::{Deserialize, Serialize};
+ use std::collections::BTreeMap;
++use std::collections::HashMap;
+ 
+ /// See Reference / Kubernetes API / Workload Resources / Pod.
+ #[derive(Clone, Debug, Serialize, Deserialize)]
+@@ -295,6 +296,9 @@ struct SecurityContext {
+ 
+     #[serde(skip_serializing_if = "Option::is_none")]
+     seccompProfile: Option<SeccompProfile>,
++
++    #[serde(skip_serializing_if = "Option::is_none")]
++    sysctls: Option<Vec<Sysctl>>,
+ }
+ 
+ /// See Reference / Kubernetes API / Workload Resources / Pod.
+@@ -307,6 +311,13 @@ struct SeccompProfile {
+     localhostProfile: Option<String>,
+ }
+ 
++/// See Reference / Kubernetes API / Workload Resources / Pod.
++#[derive(Clone, Debug, Serialize, Deserialize)]
++struct Sysctl {
++    name: String,
++    value: String,
++}
++
+ /// See Reference / Kubernetes API / Workload Resources / Pod.
+ #[derive(Clone, Debug, Serialize, Deserialize)]
+ struct PodSecurityContext {
+@@ -696,6 +707,16 @@ impl Container {
+ 
+         commands
+     }
++
++    pub fn apply_sysctls(&self, sysctls: &mut HashMap<String, String>) {
++        if let Some(securityContext) = &self.securityContext {
++            if let Some(container_sysctls) = &securityContext.sysctls {
++                for sysctl in container_sysctls {
++                    sysctls.insert(sysctl.name.clone(), sysctl.value.clone());
++                }
```

Comment:
`sysctls` is a `HashMap<String, String>`, but `container_sysctls` is a `Vec<Sysctl>`. I could add a `.map(|el| (el.name, el.value))`, but would that still be an improvement?

Reply:
```suggestion
+                sysctls.extend(container_sysctls.iter().map(|el| (el.name.clone(), el.value.clone()));
```

IMO using `extend` is preferable to an imperative for-loop because `HashMap`'s `extend` is more [optimized](https://docs.rs/hashbrown/latest/src/hashbrown/map.rs.html#6499-6514) than a simple for loop (it calls `.reserve` internally). Also it's shorter.

Reply:
Thanks, adopted.